docs/core-concepts/saturation-penalty.md

```diff
@@ -0,0 +1,80 @@
+
+When `threasholdLeaf > maxLeaf` it means that the leaf that corresponds to a penalty threshold of 85% saturation is higher than any leaf containing a position. Since `maxLeaf` is defined as `tree.highestSetLeaf`, this means that no position has been set to a leaf that is higher than the threshold leaf. In other words, no position has reached the 85% saturation threshold.
+
+On the other hand, if the a position has reached/surpassed the 85% saturation threshold, `maxLeaf > thresholdLeaf`, then the system loops through the leaves between `thresholdLeaf` and `maxLeaf` and add up all of the saturation (`satInLAssets`). The sum of all the saturation is then returned as the `satInLAssetsInPenalty`, which will be used to calculate the penalty rate.
+
+## Penalty rate calculation
+
+```
+Penalty formula:
+
+penaltyRate = ((1 - u₀) × f_interest(u₁) × totalDeposits) / (saturationInPenalty)
+
+ref: ` calcSaturationPenaltyRatePerSecondInWads` in Saturation.sol
+```
+
+The penalty rate calculation is a dynamic risk pricing system. The rate will change based on `borrow utilization` and `saturation utilization`. The motivation of this design is to ensure that LPs are properly compensated and protocol health is maintained. The relationship of the utilization metrics to penalty rate is as follows:
+
+- Borrow utilization: lower utilization -> higher penalty rate.
+- Saturation utilization: higher utilization -> higher penalty rate.
+
+The formula above defines this relationship by using (1 - u₀), where `u₀ = current borrow utilization` that results in a scaling factor. For example:
+
+1. u₀ = 10% -> (1 - 10%) = 90% scaling factor
+2. u₀ = 90% -> (1 - 90%) = 10% scaling factor
+
+**Economic incentives**
+
+The inverse relationship of borrow utilization with penalty rate is essentially because the protocal needs to balance between two objectives:
+
+1. Discourage over-saturated positions.
+  - `Low` borrow utilization and `high` saturation utilizaiton = high penalty rate
+2. Maintain system stability by preventing cascading liquidation.
+  - `High` borrow utilization and `high` saturation utiliztion = low penalty rate
+
+*Lower borrow utilizaiton*
+
+When borrow utilization is low (e.g. u₀ = 10%), this means:
+- 90% of deposited liquidity is sitting idle, earning no interest.
+- Lots of liquidity available to absorb the impact if a position is liquidated.
+
+Under the above conditions, if someone creates an over-saturated position, they are creating risk in an environment with lots of idle liquidity. Since liquidations are less of a concern goal #1 becomes more important, where a higher penalty rate can be charged to ensure that the LPs are properly compensated for the risks created. This relatively high interest rate will also discourage over-saturated positions.
```

Comment:
Once again considering positions as tracked in singular rather than in combination